### DISCUSSION TITLE
Jenkins | iOS | How to set fastlane up?

### DISCUSSION THREAD

**Discussion by pushkrajlanjekar** [fastlane/fastlane#18225] [Q&A] [answered]
Below is the content in my Fastfile

```
default_platform(:ios)
platform :ios do
  desc "FindMyBuddy IPA Maker Job"
  lane :findmybuddy_ipa_development do
    build_ios_app(
      configuration: "Debug",
      scheme: "FindMyBuddy",
      workspace: "FindMyBuddy.xcworkspace",
      include_bitcode: true,
      clean: true,
      export_method: 'development',
      output_directory: "~/Desktop", # Destination directory. Defaults to current directory.
      output_name: "fastlane_findmybuddy.ipa",
    )
  end
end
```

I am getting below error when I build my job on jenkins
**Could not find fastlane in current directory. Make sure to have your fastlane configuration files inside a folder called "fastlane". Would you like to set fastlane up?**

**Full output:**

Running as SYSTEM
[EnvInject] - Loading node environment variables.
Building in workspace /var/root/.jenkins/workspace/FindMyBuddyIPAMaker
Selected Git installation does not exist. Using Default
The recommended git tool is: NONE
using credential GIT_REPO
 > git rev-parse --is-inside-work-tree # timeout=10
Fetching changes from the remote Git repository
 > git config remote.origin.url https://github.com/pushkrajlanjekar/FindMyBuddy.git # timeout=10
Fetching upstream changes from https://github.com/pushkrajlanjekar/FindMyBuddy.git
 > git --version # timeout=10
 > git --version # 'git version 2.24.3 (Apple Git-128)'
using GIT_ASKPASS to set credentials Github
 > git fetch --tags --force --progress -- https://github.com/pushkrajlanjekar/FindMyBuddy.git +refs/heads/*:refs/remotes/origin/* # timeout=10
 > git rev-parse refs/remotes/origin/master^{commit} # timeout=10
Checking out Revision 24973a50e20f5ef9226103473b36e4f409be8d39 (refs/remotes/origin/master)
 > git config core.sparsecheckout # timeout=10
 > git checkout -f 24973a50e20f5ef9226103473b36e4f409be8d39 # timeout=10
Commit message: "Updated Fastlane Details"
 > git rev-list --no-walk 24973a50e20f5ef9226103473b36e4f409be8d39 # timeout=10
[FindMyBuddyIPAMaker] $ /bin/sh -xe /var/folders/zz/zyxvpxvq6csfxvn_n0000000000000/T/jenkins12566795792709748434.sh
+ fastlane findmybuddy_ipa_development
[20:00:23]: [33mGet started using a Gemfile for fastlane https://docs.fastlane.tools/getting-started/ios/setup/#use-a-gemfile[0m
[20:00:23]: [31mWARNING: fastlane requires your locale to be set to UTF-8. To learn more go to https://docs.fastlane.tools/getting-started/ios/setup/#set-up-environment-variables[0m
[20:00:24]: [33mCould not find fastlane in current directory. Make sure to have your fastlane configuration files inside a folder called "fastlane". Would you like to set fastlane up?[0m

Looking for related GitHub issues on fastlane/fastlane...

➡️  [33mFastlaneCore::Interface::FastlaneCrash: [!] Could not retrieve response as fastlane runs in non-interactive mode[0m
    https://github.com/fastlane/fastlane/issues/17488 [[32mclosed[0m] 11 💬
    4 weeks ago

➡️  [33mGITLAB CI: Could not retrieve response as fastlane runs in non-interactive mode[0m
    https://github.com/fastlane/fastlane/issues/17376 [[32mclosed[0m] 18 💬
    19 Dec 2020

➡️  [33mCan't update fastlane with gem[0m
    https://github.com/fastlane/fastlane/issues/17884 [[32mclosed[0m] 3 💬
    6 weeks ago

and 52 more at: https://github.com/fastlane/fastlane/search?q=Could%20not%20retrieve%20response%20as%20fastlane%20runs%20in%20non-interactive%20mode&type=Issues&utf8=✓

🔗  You can ⌘ + double-click on links to open them directly in your browser.
[20:00:26]: [31mWARNING: fastlane requires your locale to be set to UTF-8. To learn more go to https://docs.fastlane.tools/getting-started/ios/setup/#set-up-environment-variables[0m
/Library/Ruby/Gems/2.6.0/gems/fastlane-2.174.0/fastlane_core/lib/fastlane_core/ui/interface.rb:129:in `crash!': [31m[!] Could not retrieve response as fastlane runs in non-interactive mode[0m (FastlaneCore::Interface::FastlaneCrash)
	from /Library/Ruby/Gems/2.6.0/gems/fastlane-2.174.0/fastlane_core/lib/fastlane_core/ui/implementations/shell.rb:156:in `verify_interactive!'
	from /Library/Ruby/Gems/2.6.0/gems/fastlane-2.174.0/fastlane_core/lib/fastlane_core/ui/implementations/shell.rb:134:in `confirm'
	from /Library/Ruby/Gems/2.6.0/gems/fastlane-2.174.0/fastlane_core/lib/fastlane_core/ui/ui.rb:17:in `method_missing'
	from /Library/Ruby/Gems/2.6.0/gems/fastlane-2.174.0/fastlane/lib/fastlane/commands_generator.rb:362:in `ensure_fastfile'
	from /Library/Ruby/Gems/2.6.0/gems/fastlane-2.174.0/fastlane/lib/fastlane/commands_generator.rb:107:in `block (2 levels) in run'
	from /Library/Ruby/Gems/2.6.0/gems/commander-fastlane-4.4.6/lib/commander/command.rb:178:in `call'
	from /Library/Ruby/Gems/2.6.0/gems/commander-fastlane-4.4.6/lib/commander/command.rb:153:in `run'
	from /Library/Ruby/Gems/2.6.0/gems/commander-fastlane-4.4.6/lib/commander/runner.rb:476:in `run_active_command'
	from /Library/Ruby/Gems/2.6.0/gems/fastlane-2.174.0/fastlane_core/lib/fastlane_core/ui/fastlane_runner.rb:76:in `run!'
	from /Library/Ruby/Gems/2.6.0/gems/commander-fastlane-4.4.6/lib/commander/delegates.rb:15:in `run!'
	from /Library/Ruby/Gems/2.6.0/gems/fastlane-2.174.0/fastlane/lib/fastlane/commands_generator.rb:352:in `run'
	from /Library/Ruby/Gems/2.6.0/gems/fastlane-2.174.0/fastlane/lib/fastlane/commands_generator.rb:41:in `start'
	from /Library/Ruby/Gems/2.6.0/gems/fastlane-2.174.0/fastlane/lib/fastlane/cli_tools_distributor.rb:122:in `take_off'
	from /Library/Ruby/Gems/2.6.0/gems/fastlane-2.174.0/bin/fastlane:23:in `<top (required)>'
	from /usr/local/bin/fastlane:23:in `load'
	from /usr/local/bin/fastlane:23:in `<main>'
/Library/Ruby/Gems/2.6.0/gems/fastlane-2.174.0/fastlane_core/lib/fastlane_core/ui/interface.rb:129:in `crash!': Could not retrieve response as fastlane runs in non-interactive mode (FastlaneCore::Interface::FastlaneCrash)
	from /Library/Ruby/Gems/2.6.0/gems/fastlane-2.174.0/fastlane_core/lib/fastlane_core/ui/implementations/shell.rb:156:in `verify_interactive!'
	from /Library/Ruby/Gems/2.6.0/gems/fastlane-2.174.0/fastlane_core/lib/fastlane_core/ui/implementations/shell.rb:134:in `confirm'
	from /Library/Ruby/Gems/2.6.0/gems/fastlane-2.174.0/fastlane_core/lib/fastlane_core/ui/ui.rb:17:in `method_missing'
	from /Library/Ruby/Gems/2.6.0/gems/fastlane-2.174.0/fastlane/lib/fastlane/commands_generator.rb:362:in `ensure_fastfile'
	from /Library/Ruby/Gems/2.6.0/gems/fastlane-2.174.0/fastlane/lib/fastlane/commands_generator.rb:107:in `block (2 levels) in run'
	from /Library/Ruby/Gems/2.6.0/gems/commander-fastlane-4.4.6/lib/commander/command.rb:178:in `call'
	from /Library/Ruby/Gems/2.6.0/gems/commander-fastlane-4.4.6/lib/commander/command.rb:153:in `run'
	from /Library/Ruby/Gems/2.6.0/gems/commander-fastlane-4.4.6/lib/commander/runner.rb:476:in `run_active_command'
	from /Library/Ruby/Gems/2.6.0/gems/fastlane-2.174.0/fastlane_core/lib/fastlane_core/ui/fastlane_runner.rb:76:in `run!'
	from /Library/Ruby/Gems/2.6.0/gems/commander-fastlane-4.4.6/lib/commander/delegates.rb:15:in `run!'
	from /Library/Ruby/Gems/2.6.0/gems/fastlane-2.174.0/fastlane/lib/fastlane/commands_generator.rb:352:in `run'
	from /Library/Ruby/Gems/2.6.0/gems/fastlane-2.174.0/fastlane/lib/fastlane/commands_generator.rb:41:in `start'
	from /Library/Ruby/Gems/2.6.0/gems/fastlane-2.174.0/fastlane/lib/fastlane/cli_tools_distributor.rb:122:in `take_off'
	from /Library/Ruby/Gems/2.6.0/gems/fastlane-2.174.0/bin/fastlane:23:in `<top (required)>'
	from /usr/local/bin/fastlane:23:in `load'
	from /usr/local/bin/fastlane:23:in `<main>'
Build step 'Execute shell' marked build as failure
Finished: FAILURE

Path of my Fastfile in project is:

**FindMyBuddy-iOS/fastlane/Fastfile**
Other project files and folders are located inside FindMyBuddy-iOS means parallel to fastlane folder

Please let me know what steps I am missing?

**Comment by minuscorp** [comment, marked as answer]
Did you setup fastlane correctly through fastlane init? Does the Jenkins local job have the correct folder structure?

**Reply by pushkrajlanjekar** [reply]
@minuscorp 
Yes, seems like. Fastlane is correctly initiated.

The folder structure is as shown in the attached image.

<img width="209" alt="Folder Structure" src="https://user-images.githubusercontent.com/7520816/108234556-eb1dc180-716a-11eb-9d07-c266ca01b5d9.png">

**Reply by minuscorp** [reply]
Does it work locally?

**Reply by minuscorp** [reply]
It seems to me that the Jenkins node is not cloning the fastlane folder correctly because it is either out of the git repo or depth at clone time is not configured properly

**Comment by pushkrajlanjekar** [comment]
Locally using fastlane command, project is building successfully and its
creating IPA too.

But on jenkins its showing error with execute shell command of fastlane as
shown in log.

With Regards,

Mr. Pushkraj P. Lanjekar | Analyst
Fiserv, Pune.

Cell: +91 9766709921

Skype: prince.pushkraj

Bio: https://remote.com/pushkrajlanjekar


"We are mobile developers. We make smartphones even smarter."

**Comment by pushkrajlanjekar** [comment]
You were right :) thanks.. Git and Fastlane was not at same levels, its
working fine now. Thanks again.

On Wed, Feb 17, 2021 at 10:48 PM Jorge <notifications@github.com> wrote:

> It seems to me that the Jenkins node is not cloning the fastlane folder
> correctly because it is either out of the git repo or depth at clone time
> is not configured properly
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/fastlane/fastlane/discussions/18225#discussioncomment-378463>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABZMEMH2WC47S2WN2YGMBBTS7P265ANCNFSM4XYO4QLA>
> .
>


-- 
With Regards,

Pushkraj P. Lanjekar
